projects/aca-playwright-shared/src/api/upload-api.ts

```diff
@@ -0,0 +1,58 @@
+/*!
+ * Copyright © 2005-2023 Hyland Software, Inc. and its affiliates. All rights reserved.
+ *
+ * Alfresco Example Content Application
+ *
+ * This file is part of the Alfresco Example Content Application.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail. Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * The Alfresco Example Content Application is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The Alfresco Example Content Application is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import * as fs from 'fs';
+import { users } from '../base-config';
+import { ApiClientFactory } from './api-client-factory';
+
+export class UploadApi {
+  private apiService: ApiClientFactory;
+
+  constructor(userProfile?: keyof typeof users) {
+    this.apiService = new ApiClientFactory();
+    this.apiService.setUpAcaBackend(userProfile ?? 'hruser');
+  }
+
+  static async initialize(userName: string, password?: string): Promise<UploadApi> {
+    const classObj = new UploadApi();
```

Comment:
can we optimise the code like this to be 1-line?

```ts
return new UploadApi(username, password);
```

and UploadApi calling `apiService.setupAcaBackend` internally

Reply:
initialise() is an asynchronous so when we try to optymize code:
    const classObj = new SitesApi();
    await classObj.apiService.setUpAcaBackend(userName, password);
    return classObj;
into one line expression like:
return new SitesApi().apiService.setUpAcaBackend(userName, password).then(() => new SitesApi());
or
return new SitesApi().apiService.setUpAcaBackend(userName, password), new SitesApi();
we get an error with 403 access denied